src/high-level/shape.lisp

```diff
@@ -0,0 +1,30 @@
+;;;; shapes.lisp
+;;;;
+;;;; Author: Cole Scott
+
+(in-package #:magicl)
+
+;;; Shapes
+
+;; Predicates
+
+(defun valid-shape-p (shape)
+  (and (typep shape 'list)
+       (plusp (length shape))
+       (cl:every (lambda (x) (typep x 'alexandria:positive-fixnum)) shape)))
```

Comment:
Just a thought: sometimes it makes sense (perhaps after tensor contractions) to keep a zero-sized dimension. MATLAB and NumPy purge those with `squeeze`.

Reply:
Not after tensor contractions but when doing [broadcasting](https://docs.scipy.org/doc/numpy/user/basics.broadcasting.html), if we plan on supporting that at some point.

Reply:
It looks like numpy does that with dimensions of 1 rather than zero so this should still work